c10/core/Scalar.h

```diff
@@ -26,6 +27,49 @@ class C10_API Scalar {
  public:
   Scalar() : Scalar(int64_t(0)) {}
 
+  Scalar(const Scalar& rhs) : tag(rhs.tag), v(rhs.v) {
+      if (isComplex()) {
+        c10::raw::intrusive_ptr::incref(v.z);
+      }
+  }
+
+  Scalar(Scalar&& rhs) noexcept : tag(rhs.tag), v(rhs.v) {}
+
+  ~Scalar() {
+    freeComplexIfNecessary();
+  }
+
+  Scalar& operator=(const Scalar& rhs) & {
+    // TODO: Note this method is implemented differently from the lvalue assignment operator of IValue
+    freeComplexIfNecessary();
+
+    tag = rhs.tag;
+    v = rhs.v;
+
+    if (isComplex()) {
+        c10::raw::intrusive_ptr::incref(v.z);
+    }
+    return *this;
+  }
+
+  C10_ALWAYS_INLINE Scalar& operator=(Scalar&& rhs) & noexcept {
```

Comment:
any specific reason for using `C10_ALWAYS_INLINE` here? In almost all cases, the compiler knows better and forcing them to inline is a bad idea. Without this, for example, the compiler can make different decisions like say inlining it in server code but not on mobile.

Reply:
@smessmer : Got it. Originally when I start to write these move/copy methods, I followed the corresponding implementation in `IValue` and didn't reove `C10_ALWAYS_INLINE
` https://github.com/pytorch/pytorch/blob/56f8379802960b6cbd9973a02cb03efba14dfb7e/aten/src/ATen/core/ivalue.h#L197